File: Accounting_frontend/src/operation/model/operation.ts
import { EMPTY_GUID } from 'src/const';
import { ModelElementBase } from 'src/model/model-element-base';
import { OperationClass } from './operation-class';

/**
 * Операция
 */
export class Operation extends ModelElementBase {
  /**
   * Дата
   */
  public date: Date = new Date();

  /**
   * Сумма
   */
  public sum: number = 0;

  /**
   * Описание
   */
  public description: string = '';

  /**
   * Порядковый номер операции в разрезе дня
   */
  public index: number = 0;

  /**
   * Класс/тип операции, заполняется при получении на фронте
   */
  public operationClass: OperationClass = OperationClass.ContractorOperation;
}

export function operationDefault(): Operation {
  return {
    id: EMPTY_GUID,
    date: new Date(),
    sum: 0,
    description: '',
    index: -1,
    operationClass: OperationClass.ContractorOperation,
  };
}

File: AccountingDAL/Model/Operations/Operation.cs
namespace AccountingDAL.Model.Operations
{
    /// <summary>
    /// Движения по счетам
    /// </summary>
    public class Operation : ModelElementBase
    {
        /// <summary>
        /// Дата операции
        /// </summary>
        public DateTime Date { get; set; } = DateTime.Now;

        /// <summary>
        /// Сумма
        /// </summary>
        public float Sum { get; set; } = 0F;

        /// <summary>
        /// Описание
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// Порядковый номер операции в разрезе дня
        /// </summary>
        public int Index { get; set; }
    }
}
